src/main/java/com/stripe/net/LiveStripeResponseGetter.java

```diff
@@ -607,46 +614,8 @@ private static StripeResponse getMultipartStripeResponse(
 
       MultipartProcessor multipartProcessor = null;
       try {
-        multipartProcessor = new MultipartProcessor(
-            conn, boundary, ApiResource.CHARSET);
-
-        for (Map.Entry<String, Object> entry : params.entrySet()) {
-          String key = entry.getKey();
-          Object value = entry.getValue();
-
-          if (value instanceof File) {
-            File currentFile = (File) value;
-            if (!currentFile.exists()) {
-              throw new InvalidRequestException("File for key "
-                  + key + " must exist.", null, null, null, 0, null);
-            } else if (!currentFile.isFile()) {
-              throw new InvalidRequestException("File for key "
-                  + key
-                  + " must be a file and not a directory.",
-                  null, null, null, 0, null);
-            } else if (!currentFile.canRead()) {
-              throw new InvalidRequestException(
-                  "Must have read permissions on file for key "
-                      + key + ".", null, null, null, 0, null);
-            }
-            multipartProcessor.addFileField(key, currentFile.getName(),
-                new FileInputStream(currentFile));
-          } else if (value instanceof InputStream) {
-            @Cleanup InputStream inputStream = (InputStream) value;
-            if (inputStream.available() == 0) {
-              throw new InvalidRequestException(
-                "Must have available bytes to read on InputStream for key "
-                  + key + ".", null, null, null, 0, null
-              );
-            }
-            multipartProcessor.addFileField(key, "blob", inputStream);
-          } else {
-            // We only allow a single level of nesting for params
-            // for multipart
-            multipartProcessor.addFormField(key, (String) value);
-          }
-        }
-
+        multipartProcessor = new MultipartProcessor(conn, boundary, ApiResource.CHARSET);
+        encodeMultipartParams(multipartProcessor, params);
```

Comment:
Alternative design here is to have `encodeParams` as a method to multipartprocessor itself.
However, we still want to use the same `flattenParams` and `LiveRequestGetter.Parameter` within in class.